server/container/core/src/test/java/org/apache/james/core/MailImplTest.java

```diff
@@ -243,4 +250,22 @@ protected void helperTestInitialState(Mail mail) {
         assertThat(mail.getState()).as("Expecting default initial state").isEqualTo(Mail.DEFAULT);
     }
 
+    @Test
+    public void mailImplShouldBeSerializable() throws Exception {
+        MailImpl initialMail = new MailImpl();
+        initialMail.setRecipients(ImmutableList.of(MailAddressFixture.ANY_AT_JAMES));
+        initialMail.addSpecificHeaderForRecipient(
+            PerRecipientHeaders.Header.builder()
```

Comment:
It look like you don't check for a lot of property of mail that should be serialized.